examples/xonotic/cloudbuild.yaml

```
@@ -0,0 +1,54 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+  
+  #
+  # Creates the initial make + docker build platform
+  #
+  - name: "ubuntu"
+    args:
+    - "bash"
+    - "-c"
+    - "echo 'FROM gcr.io/cloud-builders/docker\nRUN apt-get install make\nENTRYPOINT [\"/usr/bin/make\"]' > Dockerfile.build"
+    waitFor: ["-"]
+
+  - name: gcr.io/cloud-builders/docker
+    id: build-make-docker
+    args: ["build", "-f", "Dockerfile.build", "-t", "make-docker", "."]
+
+  # build xonotic image to GCR
+  - name: make-docker
+    id: build
+    dir: "."
+    env:
+      - REPOSITORY=${_REPOSITORY}
+      - IMAGE_TAG=${_IMAGE_TAG}
+      - DOCKER_RUN_ARGS=--network=cloudbuild
+    args: ["-j", "1", "build"]
+  
+  # push xonotic image to GCR
+  - name: "make-docker"
+    id: push
+    waitFor:
+      - build
+    dir: "."
+    env:
+      - REPOSITORY=${_REPOSITORY}
+      - IMAGE_TAG=${_IMAGE_TAG}
+    args: ["-j", "1", "push"]
+
+substitutions:
+  _REPOSITORY: 'us-docker.pkg.dev/agones-images/examples'
```

```suggestion
  _REPOSITORY: 'us-docker.pkg.dev/${PROJECT_ID}/examples'
```

Then you can easily test it against whatever project you like, which is super nice.

---

Initially, I gave it a shot with ${PROJECT_ID}, but unfortunately, the build didn't fetch my project, leading to the following error:

 "build": docker build -f /workspace//Dockerfile --tag=us-docker.pkg.dev//examples/xonotic-example:1.1 .
"build": invalid argument "us-docker.pkg.dev//examples/xonotic-example:1.1" for "-t, --tag" flag: invalid reference format

Since I added `PROD_REPO ?= us-docker.pkg.dev/agones-images/examples`, the issue seems to be resolved.